### ISSUE TITLE
Ensure that langugae settings cannot be broken by disabling forms/submissions for all languages

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#6282]
See https://forum.pkp.sfu.ca/t/ojs-3-2-0-3-ojs-3-2-1-1-website-settings-empty-in-backend/63424/4 for details.

Currently it is possible to edit the language settings as a Journal Manager so that no languages are checked for the "Forms" and "Submissions" checkboxes. Apparently this breaks parts of the website, including settings, where there are multilingual inputs that are confused by no language options being enabled.

The language settings area should prevent users from breaking setup by disabling all languages for submissions and forms.

**Comment by henriqueramos** [comment]
Hello @NateWr, could you please check the Pull requests for this task?

[ojs](https://github.com/pkp/ojs/pull/3034)
[pkp-lib](https://github.com/pkp/pkp-lib/pull/6716)

* Evidence * 
https://user-images.githubusercontent.com/1243634/106779544-db0fc900-6625-11eb-8e30-9a7e8d98d85d.mp4

Thank you!

**Comment by henriqueramos** [comment]
Hey @NateWr, could you please take a look on the [pkp-lib](https://github.com/pkp/pkp-lib/pull/6716) pull request again? Thank you!

**Comment by NateWr** [comment]
Look good @henriqueramos! Can you redo the OJS pull request to run the tests against the latest pkp-lib changes? Once that passes this is ready for merge. :+1:

**Comment by henriqueramos** [comment]
Have fixed the conflicts @NateWr. Could you please take a look on the pull requests again?

@asmecher could you check it too, please?

**Comment by NateWr** [comment]
It looks like the tests are failing for OJS. It's the journal creation that is failing, so probably the validation checks are causing a problem. Can you reproduce the test failure when you run the tests locally?

**Comment by henriqueramos** [comment]
With the help of @NateWr we are able to figure out what happened. Sent the fix few minutes ago, but I'm waiting the CI to run to check if there's any trouble, or it's good to proceed with the normal merging flow.

**Comment by NateWr** [comment]
Looks like this is failing on a test for cases where a locale is enabled in forms and submissions but not UI: https://travis-ci.com/github/pkp/ojs/jobs/485072522#L3125. That test is https://github.com/pkp/pkp-lib/blob/main/cypress/tests/integration/Multilingual.spec.js. 

@asmecher I presume this is a use-case that we don't want to support anymore? Or was there a valid reason that a locale might be enabled at form/submission but not ui? Should we have a migration that enforces this?

**Comment by asmecher** [comment]
Eventually we do want to support forms and submissions in languages that don't have translations -- see e.g.: https://github.com/pkp/pkp-lib/issues/5000#issuecomment-749567373

**Comment by NateWr** [comment]
...uff. That means that a good chunk of the work here was unnecessary. :sob:

@asmecher at the moment, there is code in `LanguageGridHandler` which automatically removes a language from `supportedSubmissionLocales` when the language is removed from `supportedFormLocales`. Should that code be removed as well, so that each supported locale setting is completely independent?

**Comment by asmecher** [comment]
Sorry, the requirement I posted above is a new one but it would be good to start preparing for it!

Yes, I think the two settings should be independent, but the system will definitely need some exploration/testing to ensure that it behaves well when a language is supported in the forms/submissions but not the UI. And down the road we'll need to figure out a way to expose content in languages that aren't UI-supported, as the UI currently only displays localized metadata/settings in the current UI language.

**Comment by NateWr** [comment]
Ok, my bad. @henriqueramos I'm afraid I led you astray on this one. All of our hard work validating the form and submission locales against the supported locales needs to go.

So in the end we only need the following validation check in the REST API:

- When (`supportedLocales` OR `supportedFormLocales` OR `supportedSubmissionLocales`) are being validated, make sure that there is ("at least one locale" AND "this locale is enabled in the site's `supportedLocales`) .

**Comment by henriqueramos** [comment]
[New pull request at pkp-lib side](https://github.com/pkp/pkp-lib/pull/6806/files).

**Comment by henriqueramos** [comment]
@NateWr I've created a new PR at OJS repository for this issue.

https://github.com/pkp/ojs/pull/3060

**Comment by NateWr** [comment]
:tada: Merged, thanks @henriqueramos!
